.github/workflows/nightly-build.yml

```diff
@@ -1,10 +1,76 @@
 name: Nightly Build
 
+env:
+  RELEASE_BRANCH: '2025-4'
+
 on:
-  workflow_dispatch: ~
+  schedule:
+    - cron: '7 2 * * *'  # Avoid demand peaks at minute 0
+  workflow_dispatch:
+    inputs:
+      token:
+        description: 'Token with push permissions'
+        required: false
+        type: string
 
 jobs:
-  placeholder:
+  nightly-build:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        branch: [main, 2025-4]
```

Comment:
If we're going to trigger this from the regular NB bump process, the `schedule` needs to be removed and `branch` needs to be an input

Reply:
done.